koku/api/report/serializers.py

```diff
@@ -58,7 +58,37 @@ def validate_field(this, field, serializer_cls, value, **kwargs):
         (ValidationError): if field inputs are invalid
     """
     field_param = this.initial_data.get(field)
+
+    # extract tag_keys from field_params and recreate the tag_keys param
+    tag_keys = None
+    if not kwargs.get('tag_keys') and getattr(serializer_cls,
+                                              '_tagkey_support', False):
+        tag_keys = list(filter(lambda x: 'tag:' in x, field_param))
+        kwargs['tag_keys'] = tag_keys
+
     serializer = serializer_cls(data=field_param, **kwargs)
+
+    # Handle validation of multi-inherited classes.
+    #
+    # The serializer classes call super(). So, validation happens bottom-up from
+    # the BaseSerializer. This handles the case where a child class has two
+    # parents with differing sets of fields.
+    subclasses = serializer_cls.__subclasses__()
+    if subclasses and not serializer.is_valid():
```

Comment:
This is a consequence of using multiple inheritance on the OCP+AWS serializers.

When we validate, the OCP serializer doesn't know about the AWS fields and vice versa. So, for each field we're validating, we have to look up both of the OCP and AWS parent classes. Then, we run the validation against both so that we can obtain an `is_valid() == True` result from the parent class that actually knows about the field.